plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -271,7 +271,19 @@ public ResultSet executeQuery()
     @Override
     protected Optional<BiFunction<String, Long, String>> limitFunction()
     {
-        return Optional.of((sql, limit) -> sql + " LIMIT " + limit);
+        return Optional.of((sql, limit) -> {
+            if (limit > Integer.MAX_VALUE) {
+                return sql;
+            }
+            return sql + " LIMIT " + limit;
+        });
+    }
+
+    @Override
+    public boolean isLimitGuaranteed(ConnectorSession session)
+    {
+        // Note that limit exceeding Integer.MAX_VALUE gets completely ignored.
```

Comment:
This means that we'll pull all rows from Phoenix but the LIMIT will still be applied in the engine? Correct?

Reply:
Correct. Note that `isLimitGuaranteed => false` is pre-existing, i am not regressing for phoenix here.

Sadly, testing _correctnes_ would require having a table with over 2B rows, which isn't feasible in CI testing.

Reply:
I was just confirming if my understanding was correct. Looks good to me. 🙂